Automattic/wp-calypso
# Hosting Command Palette: Make View my sites always present and move filter logic together

Related to p1708964470296089/1708963140.101499-slack-C02FMH4G8

## Proposed Changes

* Make `View my sites` command always visible, even for `/sites` page.

I decided to refactor the code instead of adding a simple `.push` in the else condition, because it seems we were not adding the tracks event calypso_hosting_command_palette_command_select for that specific command.

## Testing Instructions



* Navigate to `/sites`.
* Press `cmd+k` or `ctrl+k`.
* Search "View my sites".
* Select the command.
* Observe you are redirected to the same page.

## Screencast


https://github.com/Automattic/wp-calypso/assets/779993/c7f977bc-0a62-49ae-bed2-a4d1295a1017



## Pre-merge Checklist
